tinygrad/tensor.py

```diff
@@ -497,6 +521,8 @@ def clip(self, min_, max_): return self.maximum(min_).minimum(max_)
   def abs(self): return self.relu() + (-self).relu()
   def sign(self): return self / (self.abs() + 1e-10)
   def reciprocal(self): return 1.0/self
+  def floor(self): i = self.cast(dtypes.int32).realize(); cond=i > self; return (cond * (i - 1) + (1.0 - cond) * i).cast(dtypes.int32)
```

Comment:
Methods in tensor.py can't use realize. Try contiguous here, that should prevent the casts from merging

Reply:
fixed